library/Ivoz/Core/Infrastructure/Symfony/DependencyInjection/Compiler/DomainServiceCompiler.php

```diff
@@ -90,9 +90,35 @@ protected function configureDomainEventSubscribers($fqdn)
         $service = $this->container->getDefinition($fqdn);
         $service->addTag('domain.event.subscriber');
 
-        $serviceTagName = LifecycleServiceHelper::getServiceTagByServiceFqdn($fqdn, 'on_domain_event');
-        $service->addTag($serviceTagName);
+        $interfaces = class_implements($fqdn);
+        $serviceTagName = null;
+        foreach ($interfaces as $interface) {
+            if ($interface === DomainEventSubscriberInterface::class) {
+                continue;
+            }
+
+            $isDomainEventInterface = is_subclass_of(
+                $fqdn,
+                DomainEventSubscriberInterface::class
+            );
 
+            if (!$isDomainEventInterface) {
+                continue;
+            }
+
+            $serviceTagName = LifecycleServiceHelper::getServiceTagByServiceFqdn(
+                $interface,
+                'on_domain_event'
```

Comment:
Could this be replaced with `LifecycleEventHandlerInterface::EVENT_ON_DOMAIN_EVENT` constant?